picard/tagger.py

```diff
@@ -412,7 +412,32 @@ def process(result=None, error=None):
                     self.add_files(result)
                 thread.run_task(get_files, process)
 
-        process(True, False)
+        def get_files_non_recursive():
+            listings = [f for f in os.listdir(path)]
+            if ignore_hidden:
+                listings = filter(lambda x: not is_hidden(x), listings)
+            listings = map(lambda x: os.path.join(path, x), listings)
+            files = filter(lambda x: os.path.isfile(x), listings)
+            number_of_files = len(files)
+            if number_of_files:
+                mparms = {
+                    'count': number_of_files,
+                    'directory': path,
+                }
+                self.window.set_statusbar_message(
+                    ungettext(
+                        "Adding %(count)d file from '%(directory)s' ...",
+                        "Adding %(count)d files from '%(directory)s' ...",
+                        number_of_files),
+                    mparms,
+                    translate=None)
+            return files
+
+        if config.setting['recursively_add_files']:
```

Comment:
At this point, `add_directory` only dispatches to either `process` or `get_files_non_recursive`. Imho it should instead dispatch to either `_add_directory_recursive` or `_add_directory_non_recursive` that are new methods on the tagger class. That would also help in avoiding unnecessary variables that are not needed in both cases (like `walk`).